### PR TITLE
Handle Reponse Error when building image

### PR DESCRIPTION
**Pull request by aDisplayName**
`MakeRequestForRawResponseAsync` is used when calling `IImageOperations.CreateImageAsync`. But the response error from daemon was not handled. Add the error handling before return the result.

Fix issue #555

### CODE REVIEW

**Comment by aDisplayName** [comment]
#555

**Comment by aDisplayName** [comment]
@HofmeisterAn , can you take a look?

**Review by HofmeisterAn — COMMENTED** [review]
That is great. I stumbled across this issue too. Thanks for fixing it. Can you add a small test too?

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/Docker.DotNet?pullRequest=590) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/Docker.DotNet?pullRequest=590) before we can accept your contribution.<br/><br/>:x: aDisplayName [sign now](https://cla.dotnetfoundation.org/dotnet/Docker.DotNet?pullRequest=590)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/Docker.DotNet?pullRequest=590) it.</sub>

**Comment by aDisplayName** [comment]
@HofmeisterAn , The test has been added.

**Comment by aDisplayName** [comment]
@galvesribeiro , can you help to take a look as well? Thank you.

File: test/Docker.DotNet.Tests/IImageOperationsTests.cs
Comment (HofmeisterAn) [inline test/Docker.DotNet.Tests/IImageOperationsTests.cs:75]: FYI: You do not need to `await` it. You can just return the task.

**Review by HofmeisterAn — APPROVED** [review]
LGTM. @galvesribeiro It makes sense to merge it. I know a couple of devs that run into this issue.

File: test/Docker.DotNet.Tests/IImageOperationsTests.cs
Comment (aDisplayName) [reply test/Docker.DotNet.Tests/IImageOperationsTests.cs]: The redundant async/await are removed.

**Review by HofmeisterAn — APPROVED** [review]

**Comment by aDisplayName** [comment]
Alright, so who is the "maintainer" that can approve for the workflow?

**Comment by aDisplayName** [comment]
Okay, it looks like by enabling the error handling and throw out the exception, some of the other tests utilizing the bug have failed. .... Let me figure out whether should we disable those tests:
```csharp
Docker.DotNet.Tests.ISystemOperationsTests.MonitorEventsAsync_Succeeds();
Docker.DotNet.Tests.ISystemOperationsTests.MonitorEventsFiltered_Succeeds();
```

**Comment by HofmeisterAn** [comment]
@galvesribeiro has permissions to merge it. Let's wait until he caught up. No one is maintaining the project full time.

> Let me figure out whether should we disable those tests.

We should fix the tests instead. I can take a look at it in the next days if you need some help. Just let me know.

**Comment by aDisplayName** [comment]
> @galvesribeiro has permissions to merge it. Let's wait until he caught up. No one is maintaining the project full time.
> 
> > Let me figure out whether should we disable those tests.
> 
> We should fix the tests instead. I can take a look at it in the next days if you need some help. Just let me know.

I've updated those tests.
`Docker.DotNet.Tests.ISystemOperationsTests.MonitorEventsAsync_Succeeds();` seems kind of fishy at the moment.

**Comment by HofmeisterAn** [comment]
@galvesribeiro That was a bit to early 😀, I was just replying - sorry.

I think the last commit is wrong. We should revert it and remove `CreateImageAsync` in both tests. They make no sense. The image does not exist and cannot pulled or imported. IMO the right fix would be:

```diff
diff --git a/test/Docker.DotNet.Tests/ISystemOperations.Tests.cs b/test/Docker.DotNet.Tests/ISystemOperations.Tests.cs
index 0157d54..fea6c01 100644
--- a/test/Docker.DotNet.Tests/ISystemOperations.Tests.cs
+++ b/test/Docker.DotNet.Tests/ISystemOperations.Tests.cs
@@ -111,8 +111,6 @@ namespace Docker.DotNet.Tests
                 progressMessage,
                 cts.Token);
 
-            await _dockerClient.Images.CreateImageAsync(new ImagesCreateParameters { FromImage = $"{_repositoryName}:{_tag}" }, null, progressJSONMessage, _cts.Token);
-
             await _dockerClient.Images.TagImageAsync($"{_repositoryName}:{_tag}", new ImageTagParameters { RepositoryName = _repositoryName, Tag = newTag }, _cts.Token);
 
             await _dockerClient.Images.DeleteImageAsync(
@@ -267,8 +265,6 @@ namespace Docker.DotNet.Tests
             using var cts = CancellationTokenSource.CreateLinkedTokenSource(_cts.Token);
             var task = Task.Run(() => _dockerClient.System.MonitorEventsAsync(eventsParams, progress, cts.Token));
 
-            await _dockerClient.Images.CreateImageAsync(new ImagesCreateParameters { FromImage = $"{_repositoryName}:{_tag}" }, null, new Progress<JSONMessage>());
-
             await _dockerClient.Images.TagImageAsync($"{_repositoryName}:{_tag}", new ImageTagParameters { RepositoryName = _repositoryName, Tag = newTag });
             await _dockerClient.Images.DeleteImageAsync($"{_repositoryName}:{newTag}", new ImageDeleteParameters());
```

Should I create another pull request?

**Comment by galvesribeiro** [comment]
No problem. Can you create another PR and submit the fix?

Thank you!

**Comment by HofmeisterAn** [comment]
> No problem. Can you create another PR and submit the fix?
> 
> Thank you!

Yep, I'll do. Give me a few minutes.

**Comment by aDisplayName** [comment]
@HofmeisterAn , I believe my last commit has already changed those non-existing image to the well-known 'hello-world:latest'. Or are you talking about other places?

**Comment by HofmeisterAn** [comment]
Sorry, cannot help here. Waiting for a release as well.

**Pull request merged**
